# HG changeset patch
# User prr
# Date 1400800681 25200
#      Thu May 22 16:18:01 2014 -0700
# Node ID 545facd764ecb3a0f3c328815a5c9b985bb69f90
# Parent  d6a1969c7922f7987ce00462c5f68b1e070fc4e8
8042855, PR2510: [parfait] Potential null pointer dereference in IndicLayoutEngine.cpp
Reviewed-by: bae, srl, jgodinez

diff -r d6a1969c7922 -r 545facd764ec src/share/native/sun/font/layout/IndicLayoutEngine.cpp
--- openjdk/jdk/src/share/native/sun/font/layout/IndicLayoutEngine.cpp	Thu May 22 16:17:28 2014 -0700
+++ openjdk/jdk/src/share/native/sun/font/layout/IndicLayoutEngine.cpp	Thu May 22 16:18:01 2014 -0700
@@ -151,7 +151,7 @@
     le_int32 outCharCount;
     if (fVersion2) {
         _LETRACE("v2process");
-        outCharCount = IndicReordering::v2process(&chars[offset], count, fScriptCode, outChars, glyphStorage);
+        outCharCount = IndicReordering::v2process(&chars[offset], count, fScriptCode, outChars, glyphStorage, success);
     } else {
         _LETRACE("reorder");
         outCharCount = IndicReordering::reorder(&chars[offset], count, fScriptCode, outChars, glyphStorage, &fMPreFixups, success);
diff -r d6a1969c7922 -r 545facd764ec src/share/native/sun/font/layout/IndicReordering.cpp
--- openjdk/jdk/src/share/native/sun/font/layout/IndicReordering.cpp	Thu May 22 16:17:28 2014 -0700
+++ openjdk/jdk/src/share/native/sun/font/layout/IndicReordering.cpp	Thu May 22 16:18:01 2014 -0700
@@ -1096,9 +1096,13 @@
 
 
 le_int32 IndicReordering::v2process(const LEUnicode *chars, le_int32 charCount, le_int32 scriptCode,
-                                  LEUnicode *outChars, LEGlyphStorage &glyphStorage)
+                                  LEUnicode *outChars, LEGlyphStorage &glyphStorage, LEErrorCode& success)
 {
     const IndicClassTable *classTable = IndicClassTable::getScriptClassTable(scriptCode);
+    if (classTable == NULL) {
+        success = LE_MEMORY_ALLOCATION_ERROR;
+        return 0;
+    }
 
     DynamicProperties dynProps[INDIC_BLOCK_SIZE];
     IndicReordering::getDynamicProperties(dynProps,classTable);
diff -r d6a1969c7922 -r 545facd764ec src/share/native/sun/font/layout/IndicReordering.h
--- openjdk/jdk/src/share/native/sun/font/layout/IndicReordering.h	Thu May 22 16:17:28 2014 -0700
+++ openjdk/jdk/src/share/native/sun/font/layout/IndicReordering.h	Thu May 22 16:18:01 2014 -0700
@@ -181,7 +181,7 @@
     static void adjustMPres(MPreFixups *mpreFixups, LEGlyphStorage &glyphStorage, LEErrorCode& success);
 
     static le_int32 v2process(const LEUnicode *theChars, le_int32 charCount, le_int32 scriptCode,
-        LEUnicode *outChars, LEGlyphStorage &glyphStorage);
+        LEUnicode *outChars, LEGlyphStorage &glyphStorage, LEErrorCode& success);
 
     static const FeatureMap *getFeatureMap(le_int32 &count);
 
